consensus/safety-rules/src/safety_rules_manager.rs

```diff
@@ -68,7 +68,7 @@ impl SafetyRulesManager {
             return Self::new_process(conf.server_address(), config.network_timeout_ms);
         }
 
-        let storage = storage(config);
+        let storage = storage(config).into_cached();
```

Comment:
I'm uncomfortable about using caching without check-and-set. Essentially this means that it never reads from Vault and blindly overwrites. On pre-mainnet upgrades we rely on the vault-init Terraform to reset the safety data; with this caching the current validator will just overwrite that and the upgraded validator will fail to run.

Reply:
LSR assumes it is the only entity accessing safety rules data. Could you elaborate on the vault-init / safety-rules issue.

Reply:
When doing a new genesis the safety data in Vault needs to be reset (otherwise the new validator will complain about epoch/last_voted_round/preferred_round not matching). At the moment this happens when operators apply the vault-init Terraform. This breaks the old validator, they then apply the validator Terraform which upgrades it and starts operating.